crates/codspeed/src/walltime.rs

```diff
@@ -0,0 +1,73 @@
+use std::{
+    io::Write,
+    path::{Path, PathBuf},
+};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct BenchmarkMetadata {
+    pub name: String,
+    pub uri: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RawWallTimeData {
+    #[serde(flatten)]
+    pub metadata: BenchmarkMetadata,
+    pub iter_per_round: u32,
+    pub max_time_ns: Option<u128>,
+    pub times_ns: Vec<u128>,
+}
```

Comment:
I would expect a raw walltime data to be scoped in their respective compat crates, with each compat crate in charge of translating the library raw data to a shared codspeed format.